src/handle/command.ml

```diff
@@ -376,7 +380,24 @@ let handle : (Path.t -> Sign.t) -> sig_state -> p_command ->
             in
             (Some(a), Scope.get_implicitness a)
       in
-      let ao = Option.map scope_p ao in
+      (* Show warnings if it's not a definition. *)
+      let ao = Option.map (scope_p (p_sym_trm = None)) ao in
+      (* In case of a definition, warn if variables are named but not used
+         inside the type or the definition. *)
+      let rec binders_warn ty te =
+        match ty, te with
+        | Prod(_, by), Abst(_, be) ->
+            let x, by, be = Bindlib.unbind2 by be in
+            if not (Bindlib.occur x (lift by))
+            && not (Bindlib.occur x (lift be)) then
+              wrn pos "Variable [%s] could be replaced by [_]."
+                (Bindlib.name_of x);
+            binders_warn by be
+        | _ -> ()
+      in
+      ( match ao, t with
+        | Some(a), Some(t) -> binders_warn a.elt t.elt
```

Comment:
when warn

Reply:
We want to run this code when warning in scoping have been disabled, so it would rather be `when not warn`, if I'm not mistaken.